ad-joining/register-computer/main.py

```diff
@@ -496,48 +568,45 @@ def __cleanup_computers(request):
     # a VM instance still exists, we therefore need to be careful in distinguishing
     # between the cases (1) the VM does not exist and (2) the VM is inaccessible.
 
-    # Iterate over all OUs underneath the projects OU.
+    # Iterate over all OUs underneath the projects OU or the custom OU, if specified
     projects_dn = __read_required_setting("PROJECTS_DN")
+    root_dn = os.getenv("CUSTOM_OU_ROOT_DN", projects_dn)
+    logging.info("Starting cleanup in OU '%s'" % root_dn)
+    
     result = {}
-    for ou in ad_connection.find_ou(projects_dn):
+    for ou in ad_connection.find_ou(root_dn):
         try:            
-            project_id = ou.get_name()
-
-            # Look up list of computer acconts and the zones they are located in.
+            ou_name = ou.get_dn()
+            # Look up list of computer accounts in the OU
             computer_accounts = ad_connection.find_computer(ou.get_dn())
-            zones = set([c.get_zone() for c in computer_accounts if c.get_zone() != None])
 
-            # Try to obtain the full list of instances in this project. This might
-            # fail if we have lost access to the project.
-            instance_names = gcp.project.Project(project_id).get_instance_names(zones)            
             output = {
                 "computers" : {},
                 "groups" : {}
             }
             accounts_deleted = 0
             accounts_failed = 0
 
-            logging.info("Checking for stale computer accounts in project '%s'" % project_id)
+            logging.info("Checking for stale computer accounts in OU '%s'" % ou_name)
             for computer in computer_accounts:
-                if not computer.get_instance_name():
+                if not computer.get_instance_name() or not computer.get_project_id() or not computer.get_project_id():
                     logging.info("Ignoring computer account '%s' as it lacks for GCE annotations" % computer.get_name())
```

Comment:
use debug level for this, it might get very noisy otherwise

Reply:
Agree. Fixed. Also applied when ignoring a group.